src/contextify.cc

```diff
@@ -126,27 +128,28 @@ class ContextifyContext : public ObjectWrap {
 class ContextWrap : public ObjectWrap {
 public:
     static void Init(void) {
-        NanScope();
-        Local<FunctionTemplate> tmpl = NanNew<FunctionTemplate>();
+        Nan::HandleScope scope;
+
+        Local<FunctionTemplate> tmpl = Nan::New<FunctionTemplate>();
         tmpl->InstanceTemplate()->SetInternalFieldCount(1);
-        NanAssignPersistent(functionTemplate, tmpl);
-        NanAssignPersistent(constructor, tmpl->GetFunction());
+        functionTemplate.Reset(tmpl);
+        constructor.Reset(tmpl->GetFunction());
     }
 
     static Local<Context> createV8Context(Handle<Object> jsContextify) {
-        NanEscapableScope();
-        Local<Object> wrapper = NanNew(constructor)->NewInstance();
+        Nan::EscapableHandleScope scope;
+        Local<Object> wrapper = Nan::New(constructor)->NewInstance();
 
         ContextWrap *contextWrapper = new ContextWrap();
         contextWrapper->Wrap(wrapper);
 
-        Local<Object> obj = NanNew(jsContextify);
-        NanMakeWeakPersistent(obj, contextWrapper, &weakCallback);
+        Local<Object> obj = Local<Object>::New(Isolate::GetCurrent(), jsContextify);
```

Comment:
clang generously provides:

```
../src/contextify.cc:136:29: error: no matching function for call to 'New'
        Local<Object> obj = Nan::New(jsContextify);
                            ^~~~~~~~
../node_modules/nan/nan_new.h:270:1: note: candidate function not viable: no known conversion from 'Handle<v8::Object>' (aka 'v8::Local<v8::Object>') to 'bool' for 1st argument
New(bool value) {
^
../node_modules/nan/nan_new.h:276:1: note: candidate function not viable: no known conversion from 'Handle<v8::Object>' (aka 'v8::Local<v8::Object>') to 'int32_t' (aka 'int') for 1st argument
New(int32_t value) {
^
../node_modules/nan/nan_new.h:282:1: note: candidate function not viable: no known conversion from 'Handle<v8::Object>' (aka 'v8::Local<v8::Object>') to 'uint32_t' (aka 'unsigned int') for 1st argument
New(uint32_t value) {
^
../node_modules/nan/nan_new.h:288:1: note: candidate function not viable: no known conversion from 'Handle<v8::Object>' (aka 'v8::Local<v8::Object>') to 'double' for 1st argument
New(double value) {
^
../node_modules/nan/nan_new.h:294:1: note: candidate function not viable: no known conversion from 'Handle<v8::Object>' (aka 'v8::Local<v8::Object>') to 'const std::string' (aka 'const basic_string<char>') for 1st argument
New(std::string const& value) {  // NOLINT(build/include_what_you_use)
^
../node_modules/nan/nan_new.h:312:1: note: candidate function not viable: no known conversion from 'Handle<v8::Object>' (aka 'v8::Local<v8::Object>') to 'const char *' for 1st argument
New(const char * value) {
^
../node_modules/nan/nan_new.h:318:1: note: candidate function not viable: no known conversion from 'Handle<v8::Object>' (aka 'v8::Local<v8::Object>') to 'const uint16_t *' (aka 'const unsigned short *') for 1st argument
New(const uint16_t * value) {
^
../node_modules/nan/nan_new.h:324:1: note: candidate function not viable: no known conversion from 'Handle<v8::Object>' (aka 'v8::Local<v8::Object>') to 'v8::String::ExternalStringResource *' for 1st argument
New(v8::String::ExternalStringResource * value) {
^
../node_modules/nan/nan_new.h:330:1: note: candidate function not viable: no known conversion from 'Handle<v8::Object>' (aka 'v8::Local<v8::Object>') to 'ExternalOneByteStringResource *' (aka 'v8::String::ExternalOneByteStringResource *') for 1st argument
New(ExternalOneByteStringResource * value) {
^
../node_modules/nan/nan_new.h:207:1: note: candidate template ignored: couldn't infer template argument 'T'
New(A0 arg0) {
^
../node_modules/nan/nan_new.h:237:1: note: candidate template ignored: couldn't infer template argument 'T'
New( FunctionCallback callback
^
../node_modules/nan/nan_new.h:245:1: note: candidate template ignored: couldn't infer template argument 'T'
New( FunctionCallback callback
^
../node_modules/nan/nan_new.h:259:25: note: candidate template ignored: could not match 'Persistent' against 'Local'
    inline v8::Local<T> New(v8::Persistent<T, M> const& p);
                        ^
../node_modules/nan/nan_new.h:264:21: note: candidate template ignored: could not match 'Persistent' against 'Local'
inline v8::Local<T> New(Persistent<T, M> const& p);
                    ^
../node_modules/nan/nan_new.h:266:21: note: candidate template ignored: could not match 'Global' against 'Local'
inline v8::Local<T> New(Global<T> const& p);
                    ^
../node_modules/nan/nan_new.h:201:1: note: candidate function template not viable: requires 0 arguments, but 1 was provided
New() {
^
../node_modules/nan/nan_new.h:213:1: note: candidate function template not viable: requires 2 arguments, but 1 was provided
New(A0 arg0, A1 arg1) {
^
../node_modules/nan/nan_new.h:219:1: note: candidate function template not viable: requires 3 arguments, but 1 was provided
New(A0 arg0, A1 arg1, A2 arg2) {
^
../node_modules/nan/nan_new.h:225:1: note: candidate function template not viable: requires 4 arguments, but 1 was provided
New(A0 arg0, A1 arg1, A2 arg2, A3 arg3) {
^
../node_modules/nan/nan_new.h:300:1: note: candidate function not viable: requires 2 arguments, but 1 was provided
New(const char * value, int length) {
^
../node_modules/nan/nan_new.h:306:1: note: candidate function not viable: requires 2 arguments, but 1 was provided
New(const uint16_t * value, int length) {
^
../node_modules/nan/nan_new.h:336:1: note: candidate function not viable: requires 2 arguments, but 1 was provided
New(v8::Local<v8::String> pattern, v8::RegExp::Flags flags) {
^
1 error generated.
```
(Wow, what a great error message compared to what I'd expect from GCC)